test/elixir_analyzer/exercise_test/two_fer_test.exs

```diff
@@ -0,0 +1,260 @@
+    test_exercise_analysis "multiple implementations of two_fer/1 don't count as auxiliary functions",
+      comments_exclude: [Constants.two_fer_use_of_aux_functions()] do
+      defmodule TwoFer do
+        def two_fer("foo") do
+          "One for foo, one for me"
+        end
+
+        def two_fer(name) do
+          "One for #{name}, one for me"
+        end
+      end
+    end
+
+    test_exercise_analysis "refer there are other functions than two_fer/1",
+      comments_include: [Constants.two_fer_use_of_aux_functions()],
+      status: :refer do
+      [
+        defmodule TwoFer do
+          def two_fer(name \\ "you") do
+            "One for #{name}, one for me"
+          end
+
+          defp foo() do
+          end
+        end,
+        defmodule TwoFer do
+          def two_fer(name \\ "you") do
+            "One for #{name}, one for me"
+          end
+
+          defp foo(), do: nil
+        end,
+        defmodule TwoFer do
+          def two_fer(name \\ "you") do
+            "One for #{name}, one for me"
+          end
+
+          defp foo(_a), do: nil
+        end,
+        defmodule TwoFer do
+          def two_fer(name \\ "you") do
+            "One for #{name}, one for me"
+          end
+
+          defp foo(_a, _b), do: nil
+        end
+        #        defmodule TwoFer do
+        #          def two_fer(name \\ "you") do
+        #            "One for #{name}, one for me"
+        #          end
+        #
+        #          def foo(_a, _b), do: nil
+        #        end,
+        #        defmodule TwoFer do
+        #          def foo(_a, _b, _c, _d), do: nil
+        #
+        #          def two_fer(name \\ "you") do
+        #            "One for #{name}, one for me"
+        #          end
+        #        end,
+        #        defmodule TwoFer do
+        #          def foo(_a) do
+        #          end
+        #
+        #          def two_fer(name \\ "you") do
+        #            "One for #{name}, one for me"
+        #          end
+        #        end
+      ]
+    end
+  end
+
+  describe "string interpolation" do
+    test_exercise_analysis "doesn't allow string concatenation",
+      comments_include: [Constants.two_fer_use_string_interpolation()],
+      status: :disapprove do
+      defmodule TwoFer do
+        def two_fer(name \\ "you") when is_binary(name) do
+          "One for " <> name <> ", one for me"
+        end
+      end
+    end
+  end
+
+  #  describe "FunctionClauseError" do
+  #    test_exercise_analysis "doesn't allow raising FunctionClauseError explicitly",
+  #      comments_include: [Constants.solution_raise_fn_clause_error()],
+  #      status: :disapprove do
+  #      defmodule TwoFer do
+  #        def two_fer(name \\ "you") do
+  #          if is_bitstring(name) do
+  #            "One for #{name}, one for me"
+  #          else
+  #            raise FunctionClauseError
+  #          end
+  #        end
+  #      end
+  #    end
+  #  end
```

Comment:
This test currently fails and I think that might be a bug in the analyzer... it fails because the status of the analysis is `:refer` instead of `:disapprove`. I think the `:refer` status comes from the feature `"uses function level guard"`, but I expected `:disapprove` to always take priority over `:refer`. Since `:refer` means it's impossible to know if the solution is optimal or not, if two feature checks fail, one with `on_fail` of `:refer`, and one `:disapprove`, it _can_ be known for sure that the solution is not optimal.

Reply:
Not sure if a bug, I think it was a how the spec was defined for V2.  I think you are right though, for V3, :disapprove should take precidence over :refer